src/envs/behavior.py

```diff
@@ -110,8 +112,9 @@ def __init__(self) -> None:
                 self._options.add(option)
 
     def simulate(self, state: State, action: Action) -> State:
-        assert state.simulator_state is not None
-        self.igibson_behavior_env.task.reset_scene(state.simulator_state)
+        if not state.allclose(self.current_ig_state_to_state()):
```

Comment:
I feel like there's not too much point in this, since it'll change 2 lines into 1 line, but we'll have to add a new class function that itself calls `load_checkpoint_states`, which is now its own separate function. I'm more inclined to keep things this way, but take a look at the new refactor and let me know if you still feel we should implement this change and I'm happy to do it!

Reply:
got it, makes sense. I'll defer to your judgement!